src/topics/group/group.api.test.ts

```diff
@@ -1,40 +1,55 @@
+import "reflect-metadata";
 import request from "supertest";
-import app from "../../api/app";
-import resetTestInfrastructure from "../../infrastructure/test-infrastructure";
-import testGroups from "./test-groups";
+import { DependencyContainer } from "tsyringe";
+import { getFastify } from "../../api/app";
+import { getTestGroups } from "./test-groups";
+import { FastifyInstance } from "fastify";
+import { getMemoryContainer } from "../../infrastructure";
+import { Group } from "./group";
 
 describe("test groups api", () => {
+  let fastify: FastifyInstance;
+  let container: DependencyContainer;
+  let testGroups: { [name: string]: Group };
+
+  beforeAll(() => {
+    container = getMemoryContainer();
+  });
+
   beforeEach(async () => {
-    await resetTestInfrastructure();
+    container.clearInstances();
+    fastify = getFastify(false, {}, container);
+    await fastify.ready();
+    testGroups = getTestGroups(container);
   });
 
-  test("Should respond 400 without groupName", async () => {
-    const response = await request(app).get("/groups");
+  it("Should respond 400 without groupName", async () => {
+    const response = await request(fastify.server).get("/groups");
     expect(response.statusCode).toBe(400);
   });
 
-  test("Should get empty items", async () => {
-    const response = await request(app).get(
+  it("Should get empty items", async () => {
+    const response = await request(fastify.server).get(
       `/groups?groupName=${testGroups.group1_0.name}`
     );
     expect(response.statusCode).toBe(200);
     expect(response.body.items).toEqual([]);
   });
 
-  test("Should store groups and get all", async () => {
-    await testGroups.group1_0.save();
-    await testGroups.group1_1.save();
-    const response = await request(app).get(
+  it("Should store groups and get all", async () => {
```

Comment:
I think we could create a specific "describe" for all search and have a before for each of the test of this describe. This avoid to re-create all group each time.. It's heavy otherwise